src/ios/SentryCordova.m

```diff
@@ -54,6 +58,60 @@ - (void)startWithOptions:(CDVInvokedUrlCommand *)command {
   [self.commandDelegate sendPluginResult:result callbackId:command.callbackId];
 }
 
+- (SentryOptions *_Nullable)createOptionsWithDictionary:(NSDictionary *_Nonnull)options
+                                         error: (NSError *_Nonnull *_Nonnull) errorPointer
+{
+    SentryBeforeSendEventCallback beforeSend = ^SentryEvent*(SentryEvent *event) {
+        // We don't want to send an event after startup that came from a Unhandled JS Exception of Cordova
+        // Because we sent it already before the app crashed.
+        if (nil != event.exceptions.firstObject.type &&
+            [event.exceptions.firstObject.type rangeOfString:@"Unhandled JS Exception"].location != NSNotFound) {
+            NSLog(@"Unhandled JS Exception");
+            return nil;
+        }
+
+        [self setEventOriginTag:event];
+
+        return event;
+    };
+
+    NSMutableDictionary * mutableOptions =[options mutableCopy];
+    [mutableOptions setValue:beforeSend forKey:@"beforeSend"];
+
+    // remove performance traces sample rate and traces sampler since we don't want to synchronize these configurations
+    // to the Native SDKs.
+    // The user could tho initialize the SDK manually and set themselves.
+    [mutableOptions removeObjectForKey:@"tracesSampleRate"];
+    [mutableOptions removeObjectForKey:@"tracesSampler"];
+    [mutableOptions removeObjectForKey:@"enableTracing"];
+
+    SentryOptions *sentryOptions = [[SentryOptions alloc] initWithDict:mutableOptions didFailWithError:errorPointer];
+    if (*errorPointer != nil) {
+        return nil;
+    }
```

Comment:
l: Apples suggested way to track error is to check the result, the pointer may be set already. I dont think it would be the case here, but its good to keep a good habit.

You could also log the error in the terminal here.

```suggestion
    if (sentryOptions == nil) {
        //Log the error
        return nil;
    }
```